### PR TITLE
Improve Font Awesome building and handling (update to 6.5.2)

### LINKED ISSUE #1090: Fontawesome 6.1.1 to 6.5.1 for remove Font console error
### Winter CMS Build

dev-develop

### PHP Version

8.1

### Database engine

MySQL/MariaDB

### Plugins installed

_No response_

### Issue description

In Backend, Fontawesome cause error message.
Up to Fontawesome 6.5.1 resolve Font console error :
`downloadable font: Glyph bbox was incorrect (glyph ids 5 9 10 12 13 15 16 33 49 59 60 61 72 76 81 83 84 85 88 91 92 97 101 102 103 104 106 107 109 118 120 121 122 124 134 135 136 137 139 141 143 145 146) (font-family: "Font Awesome 6 Free" style:normal weight:300 stretch:100 src index:0) source: /webfonts/xxxxxxx.woff2`

### Steps to replicate

Font Error message in logs browser console from Backend

### Workaround

_No response_

### PR DESCRIPTION
**Pull request by bennothommo**
- Bring in Font Awesome through NPM, allowing us to simply run `php artisan mix:update` to get latest Font Awesome version.
- Build core styles and Font Awesome through Winter Mix.
- Copy webfonts from Font Awesome vendor directly dynamically into Storm UI assets.

Fixes https://github.com/wintercms/winter/issues/1090

### CODE REVIEW

**Comment by bennothommo** [comment]
The Workshop theme now has a page that allows you to review the available Font Awesome icons. You can install it then navigate to **CMS** -> **Font Awesome Icons**.

File: modules/backend/formwidgets/permissioneditor/assets/css/permissioneditor.css [hunk 1]
```diff
@@ -4,7 +4,7 @@
 .permissioneditor table{width:100%}
 .permissioneditor table th{padding:30px 4px 8px 4px;color:#2a3e51;font-weight:normal;border-bottom:1px solid #dbe1e3}
 .permissioneditor table th.tab{font-size:13px}
-.permissioneditor table th.permission-type{text-transform:uppercase;font-size:11px;text-align:center;cursor:pointer}
+.permissioneditor table th.permission-type{text-transform:uppercase;font-size:11px;text-align:center}
```
Comment (LukeTowers) [inline modules/backend/formwidgets/permissioneditor/assets/css/permissioneditor.css]: Why was this removed?

**Comment by LukeTowers** [comment]
@bennothommo this might cause problems with how I load in the fonts on the frontend for use in my EasyForms plugin (as well as a simplified version for just icons):

`plugins/cadets/core/assets/less/select.less`: (used on https://cadets.site/)
```less
@base_path: "../../../../..";
// @base_path: "../../../../../Core/winter";

@winterAssets: "@{base_path}/modules/backend/assets/less";
@stormAssets: "@{base_path}/modules/system/assets/ui/less";

@import "@{winterAssets}/core/boot.less";
@font-path: "/modules/system/assets/ui/font";
@image-path: "/modules/system/assets/ui/images";
@icon-font-path: @font-path;

.unit-selector {
    @import "@{stormAssets}/../icons.less";
}

@import "@{stormAssets}/select.less";
```

**Comment by LukeTowers** [comment]
Could we come up with a simplified way to include the icon assets in the frontend (and while we're at it maybe the select2 assets as well)?

File: modules/backend/formwidgets/permissioneditor/assets/css/permissioneditor.css (hunk 1 above)
Comment (bennothommo) [reply modules/backend/formwidgets/permissioneditor/assets/css/permissioneditor.css]: It's not in the source LESS file. I probably ran the old util compile at some point.

Comment (LukeTowers) [reply modules/backend/formwidgets/permissioneditor/assets/css/permissioneditor.css]: Should we add it back in?

Comment (bennothommo) [reply modules/backend/formwidgets/permissioneditor/assets/css/permissioneditor.css]: Evidentally, it was never in there. @marvindurot added that CSS in https://github.com/wintercms/winter/pull/936 but we forgot to get it included in the LESS file. I'll re-add and re-gen.

**Comment by bennothommo** [comment]
@LukeTowers why are you including the entire icon LESS file inside the `.unit-selector` class selector? Are you intending for icons to only show within that class?

As to how to include the icon assets easily in the frontend - the icons CSS file is already split off from the rest of Storm, and is accessible at `modules/system/assets/ui/icons.css`. It would be trivial to just include that in any theme. The Select2 CSS is currently merged in with the rest of the Storm UI, but I suppose you could split off its CSS as well.

**Comment by LukeTowers** [comment]
After looking at it again, it's fine for https://cadets.site to just include the icons.css directly, but the reason I took that approach with the EasyForms plugin is because I didn't want to cause any conflicts with existing icon systems in themes. It's not the end of the world though, if it causes a problem for someone then I can always just make it optionally include that CSS and they can figure out how they want icons to work in their theme themselves.
